src/common/setting.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_COMMON_SETTING_H
+#define BITCOIN_COMMON_SETTING_H
+
+#include <common/args.h>
+
+#include <univalue.h>
+
+namespace common {
+//! State representing a setting that is unset
+struct Unset {};
+//! State representing a setting that is enabled without a value ("-setting")
+struct Enabled {};
+//! State representing a setting that is disabled ("-nosetting")
+struct Disabled {};
+
+struct SettingOptions {
+    bool legacy{false};
+    bool debug_only{false};
+    bool network_only{false};
+    bool sensitive{false};
+    bool disallow_negation{false};
+    bool disallow_elision{false};
+};
+
+template<size_t N, typename T=char[N]>
+struct StringLiteral {
+    constexpr StringLiteral(const char (&str)[N]) {
+        std::copy_n(str, N, value);
+    }
+    constexpr StringLiteral(std::nullptr_t) {
+    }
+    T value{};
+};
+
+StringLiteral(std::nullptr_t) -> StringLiteral<1, std::nullptr_t>;
+} // namespace common
+
+#include <common/setting_internal.h>
+
+namespace common {
+//! Setting template class used to declare compile-time Setting types which are
+//! used to register and retrieve settings.
+template<StringLiteral summary, typename T, SettingOptions options = SettingOptions{}, StringLiteral help = nullptr, auto help_fn = nullptr, auto default_fn = nullptr, auto get_fn = nullptr, OptionsCategory category = OptionsCategory::OPTIONS>
+struct Setting {
+    using value_t = internal::RemoveOptional<T>::type;
```

Comment:
I hinted at the IDE complaining since I could exactly find out why this was necessary - if you can't see the value in it either, please resolve this comment.